dissect/target/plugins/os/windows/notifications.py

```diff
@@ -63,11 +198,149 @@ def __init__(self, target):
+        user: RecordDescriptor,
+    ) -> AppDBChunkRecord:
+        push_timestamp1 = None
+        if ts := chunk.PushDescriptor.Timestamp1:
+            push_timestamp1 = wintimestamp(ts)
+        push_timestamp2 = None
+        if ts := chunk.PushDescriptor.Timestamp1:
+            push_timestamp2 = wintimestamp(ts)
+        push_uri = chunk.PushDescriptor.Uri.split(b"\x00")[0]
+        push_uri = push_uri.decode("utf-8", errors="replace")
+
+        badge_ts = None
+        if ts := chunk.BadgeContent.Timestamp:
+            badge_ts = wintimestamp(ts)
+        badge_data = chunk.BadgeContent.Data.decode("utf-8", errors="replace")
+
+        return AppDBChunkRecord(
+            last_notification_date=last_notification_date,
+            next_notification_id=chunk.Header.NextNotificationId,
+            push_ts1=push_timestamp1,
+            push_ts2=push_timestamp2,
+            push_uri=push_uri,
+            badge_id=chunk.BadgeContent.Id,
+            badge_ts=badge_ts,
+            badge_data=badge_data,
+            _target=self.target,
+            _user=user,
+        )
+
+    def _get_appdb_tile_records(self, chunk: c_appdb.Chunk, user: RecordDescriptor) -> list[AppDBTileRecord]:
+        tile_records = []
+        num_tiles = len(chunk.Tiles)
+
+        for tile_no in range(num_tiles):
+            tile = chunk.Tiles[tile_no]
+
+            if tile.UniqId:
+                tile_ts = None
+                if ts := tile.TS:
+                    tile_ts = wintimestamp(ts)
+                tile_expiry_ts = None
+                if ts := tile.TS_Expiry:
+                    tile_expiry_ts = wintimestamp(ts)
+                name = tile.Name.strip("\x00")
+
+                xml_size = tile.ContentLength
+                tile_xml = chunk.TileXml[tile_no][:xml_size]
+                tile_xml = tile_xml.decode("utf-8", errors="replace")
+
+                tile_record = AppDBTileRecord(
+                    id=tile.UniqId,
+                    ts=tile_ts,
+                    expiry_ts=tile_expiry_ts,
+                    type=tile.Type,
+                    index=tile.Index,
+                    name=name,
+                    content=tile_xml,
+                    _target=self.target,
+                    _user=user,
+                )
+
+                tile_records.append(tile_record)
+        return tile_records
+
+    def _get_appdb_toast_records(self, chunk: c_appdb.Chunk, user: RecordDescriptor) -> list[AppDBToastRecord]:
+        toast_records = []
+        num_toasts = len(chunk.Toasts)
+
+        for toast_no in range(num_toasts):
+            toast = chunk.Toasts[toast_no]
+
+            if toast.UniqId:
+                toast_ts = None
+                if ts := toast.TS:
+                    toast_ts = wintimestamp(ts)
+                toast_expiry_ts = None
+                if ts := toast.TS_Expiry:
+                    toast_expiry_ts = wintimestamp(ts)
+                name1 = toast.Name1.strip("\x00")
```

Comment:
Just a general suggestion... I notice that thi NUL-byte handling is becoming a very common thing (41 matches in target alone!), maybe we should add a convenience function in cstruct for this? I.e. cstr(...) ? because apparently we want to extract the c-string.

Reply:
There is such a function in cstruct for all relevant types: `_read_0()` it seems though that it is not exposed.